### PR TITLE
Extvlcopt support matrix

### PR DESCRIPTION
**Pull request by phunkyfish**
v4.1.0
- Support EXTVCOPT in m3u

### CODE REVIEW

**Comment by MilhouseVH** [comment]
This has been in test builds for almost a month (since `#0630`) and no reported issues - time to merge?

**Comment by phunkyfish** [comment]
Yes, that would be great. There is also backport PR #245

**Comment by phunkyfish** [comment]
@ksooo could you help with getting these PRs merged?

**Comment by MilhouseVH** [comment]
Ping...

**Comment by ksooo** [comment]
> could you help with getting these PRs merged?

 Sure, if you could explain to me what this change is good for. The PR description tells me nothing. :-/

**Comment by phunkyfish** [comment]
Completely fair. 

We added code to the FFMPeg demux to support setting the program ID. The PR allows this to be set via M3u8 for simpleiptv.

**Comment by phunkyfish** [comment]
I could now actually merge these myself once you are happy ;)

**Comment by ksooo** [comment]
I spotted some strings that could be `const`. :-)

> I could now actually merge these myself once you are happy ;)

 Let's follow the official process also here. Once you have the formal approval of another guy with some knowledge you can merge it, like we've done/should also do in future with vuplus.

**Comment by phunkyfish** [comment]
> I spotted some strings that could be `const`. :-)
> 

Will I ever learn...

> 
> Let's follow the official process also here. Once you have the formal approval of another guy with some knowledge you can merge it, like we've done/should also do in future with vuplus.

Sure, makes sense.

**Comment by phunkyfish** [comment]
Jenkins looks to be stalled on this.

File: src/PVRIptvData.cpp
Comment (ksooo) [inline src/PVRIptvData.cpp]: missing blank before `pos`

File: src/PVRIptvData.cpp
Comment (ksooo) [inline src/PVRIptvData.cpp]: missing blank before and after `+`

File: src/PVRIptvData.cpp
Comment (ksooo) [inline src/PVRIptvData.cpp]: no blank before `&`, but after

File: src/PVRIptvData.h
Comment (ksooo) [inline src/PVRIptvData.h]: no blank before &, but after

File: src/PVRIptvData.h
Comment (ksooo) [inline src/PVRIptvData.h]: no blank before &, but after

BTW: All this methods being `virtual` makes zero sense. But this is not for this PR.

Comment (phunkyfish) [reply src/PVRIptvData.h]: Will give it a refactor at some point

Comment (ksooo) [reply src/PVRIptvData.h]: 👍 that would be highly appreciated

**Review by ksooo — APPROVED** [review]
Thank you

**Comment by phunkyfish** [comment]
jenkins build this please

**Comment by phunkyfish** [comment]
I’m not sure why iOS is failing. It appears to build correctly.

**Comment by ksooo** [comment]
> I’m not sure why iOS is failing. It appears to build correctly.

 Me neither. @Rechi do you have an explanation?

**Comment by Rechi** [comment]
Because https://github.com/xbmc/xbmc/pull/16039 changed the zip file name.

**Comment by ksooo** [comment]
> Because xbmc/xbmc#16039 changed the zip file name.

 Okay, and what needs to be done to fix the build issue?

**Comment by Rechi** [comment]
Change the ZIP name / path for iOS from `<ADDON-ID>+darwin_embedded-<ARCH>/<ADDON-ID>-<VERSION>.zip` back to `<ADDON-ID>+ios-<ARCH>/<ADDON-ID>-<VERSION>.zip`.

**Comment by ksooo** [comment]
> Change the ZIP name / path for iOS from <ADDON-ID>+darwin_embedded-<ARCH>/<ADDON-ID>-<VERSION>.zip back to <ADDON-ID>+ios-<ARCH>/<ADDON-ID>-<VERSION>.zip.

 Sorry for my dumb questions: And **where** does this need to be fixed?

**Comment by Rechi** [comment]
In the same repo as the PR, which broke it, https://github.com/xbmc/xbmc/.

**Comment by ksooo** [comment]
> In the same repo as the PR, which broke it, https://github.com/xbmc/xbmc/.

 Over my head, sorry. **Who** could/must do this and **where exactly**?

**Comment by Rechi** [comment]
I would wait for the PR author to have a look, but you can fix it too.

https://github.com/xbmc/xbmc/blob/43d3c7eb52c60a3f382b8e00d32abcdee1d5858e/cmake/scripts/common/AddonHelpers.cmake#L22-L24 defines the output name / path. So you can start from there.

**Comment by phunkyfish** [comment]
This would be @fuzzard I guess. Related to tvOS. I think he’s away for a week though.

**Comment by phunkyfish** [comment]
@SylvainCecchetto, is this something you could look at? You’re the only other cmake whiz I know, plus you have some knowledge of the PR. Totally understand if you are busy with your thesis.

**Comment by phunkyfish** [comment]
@Rechi 

From his holiday fuzzard sent this on. Can't test it currently, for some reason I can't build iOS (local issues on laptop).

```
--- a/cmake/scripts/common/PrepareEnv.cmake    2019-07-23 19:32:19.000000000 +1000
+++ b/cmake/scripts/common/PrepareEnv.cmake    2019-08-11 08:43:59.000000000 +1000
@@ -46,9 +46,9 @@
   endif()
 elseif(CORE_SYSTEM_NAME STREQUAL darwin_embedded)
   if (CPU MATCHES armv7)
-    set(PLATFORM_TAG ${PLATFORM_TAG}-armv7)
+    set(PLATFORM_TAG ${CORE_PLATFORM_NAME_LC}-armv7)
   elseif (CPU MATCHES arm64)
-    set(PLATFORM_TAG ${PLATFORM_TAG}-aarch64)
+    set(PLATFORM_TAG ${CORE_PLATFORM_NAME_LC}-aarch64)
   else()
     message(FATAL_ERROR "Unsupported architecture")
   endif()
```

**Comment by fuzzard** [comment]
Had time to do a quick test, CORE_PLATFORM_NAME_LC isnt available to prepareenv.cmake, however CORE_PLATFORM_NAME is. change that in the above diff, and i think you might be right.

**Comment by phunkyfish** [comment]
PR for patch: https://github.com/xbmc/xbmc/pull/16475

**Comment by phunkyfish** [comment]
We're back in action on Matrix at least. Thank @fuzzard and @Rechi 

@ksooo had some time so added a bunch of refactoring, if you's like this moved to a new PR just say so.

**Comment by phunkyfish** [comment]
@MilhouseVH could you add this PR to the testbuilds?

EDIT: ignore me. Sure it’s already in there.

**Comment by phunkyfish** [comment]
Will merge this tomorrow assuming no reported issues with test builds.

**Pull request merged**
